src/freenet/clients/http/WelcomeToadlet.java

```diff
@@ -265,6 +265,18 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
             writeHTMLReply(ctx, 200, "OK", pageNode.generate());
             request.freeParts();
             bucket.free();
+            return;
+        } else if (request.isPartSet("key")) {
+            if(!ctx.checkFormPassword(request)) return;
+            String key;
+            try {
+              key = URLDecoder.decode(new FreenetURI(request.getPartAsStringFailsafe("key", 256)).toURI("/").toString(), false);
+            } catch (Exception e) {
+              sendErrorPage(ctx, "Invalid URI", "This URI is invalid!", e);
```

Comment:
I don't care much for the more obscure (internal) error messages, but this one can easily be triggered by the user mistyping stuff. This is user interface stuff after all.

I won't object to having this merged without being l10n'ed, but if you find the time, please do so.